File: cn_indian_payroll/cn_indian_payroll/report/employee_reco/employee_reco.py
import frappe

def get_salary_slips(filters=None):
    if filters is None:
        filters = {}

    conditions = {"docstatus": ["in", [0, 1]]}

    if filters.get("select_month"):
        conditions["custom_month"] = filters["select_month"]

    if filters.get("payroll_period"):
        conditions["custom_payroll_period"] = filters["payroll_period"]

    if "custom_month" in conditions and "custom_payroll_period" in conditions:
        frappe.msgprint(str(conditions))  

def execute(filters=None):
    columns = []

    data = get_salary_slips(filters)

    return columns, data

File: cn_indian_payroll/cn_indian_payroll/report/employee_reco/employee_reco.js
// Copyright (c) 2024, Hybrowlabs technologies and contributors
// For license information, please see license.txt

frappe.query_reports["Employee Reco"] = {
	"filters": [

		{
            "label": "Select Month",
            "fieldname": "select_month",
            "fieldtype": "Select",
            "options": ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"],
            "width": 200
        },

		{
            "label": "Select Payroll Period",
            "fieldname": "payroll_period",
            "fieldtype": "Link",
            "options": "Payroll Period",
            "width": 200
        },


	]
};
